tests/test_bashcomplete.py

```diff
@@ -125,15 +125,16 @@ def asub(asub_opt):
 
     @cli.command('bsub')
     @click.option('--bsub-opt')
-    @click.argument('arg', type=click.Choice(['arg1', 'arg2']))
+    @click.argument('arg', type=click.Choice(['arg1', 'arg2']), required=True)
     def bsub(bsub_opt, arg):
         pass
 
     assert list(get_choices(cli, 'lol', [], '-')) == ['--cli-opt']
     assert list(get_choices(cli, 'lol', [], '')) == ['asub', 'bsub']
     assert list(get_choices(cli, 'lol', ['asub'], '-')) == ['--asub-opt']
     assert list(get_choices(cli, 'lol', ['asub'], '')) == ['bsub']
-    assert list(get_choices(cli, 'lol', ['bsub'], '')) == ['arg1', 'arg2', 'asub']
```

Comment:
In general yes, though note that bsub *requires* the "arg" argument, so we know that the smart thing to do is to complete that before offering subcommands.